code/workspaces/client-api/src/schema/resolvers.js

```diff
@@ -21,17 +20,16 @@ const { elementPermissions: { SETTINGS_READ, SETTINGS_EDIT } } = permissionTypes
 const { SYSTEM_INSTANCE_ADMIN } = permissionTypes;
 const { READY } = statusTypes;
 
-const DATALAB_NAME = config.get('datalabName');
 const PROJECT_KEY = 'project';
 
 const resolvers = {
   Query: {
     status: () => () => `GraphQL server is running version: ${version}`,
     dataStorage: (obj, args, { user, token }) => projectPermissionWrapper(args, STORAGE_LIST, user, () => storageService.getAllProjectActive(args.projectKey, token)),
     dataStore: (obj, args, { user, token }) => projectPermissionWrapper(args, STORAGE_OPEN, user, () => storageService.getById(args.projectKey, args.id, token)),
-    stack: (obj, { id }, { user, token }) => permissionChecker(STACKS_OPEN, user, () => stackService.getById({ user, token }, id)),
-    stacks: (obj, args, { user, token }) => permissionChecker(STACKS_LIST, user, () => stackService.getAll({ user, token })),
-    stacksByCategory: (obj, args, { user, token }) => permissionChecker(STACKS_LIST, user, () => stackService.getAllByCategory(PROJECT_KEY, args.category, { user, token })),
+    stack: (obj, args, { user, token }) => projectPermissionWrapper(args, STACKS_OPEN, user, () => stackService.getById(args.id, { user, token })),
```

Comment:
Isn't the signature now `getById(projectKey, id, { token })`?

Reply:
No, weirdly it isn't, the underlying endpoint gets all stacks....but isn't used. The plan is to delete this after validating that it isn't used.

Reply:
You're absolutely right. Will fix

Reply:
Fixed